cicecore/cicedynB/infrastructure/ice_grid.F90

```diff
@@ -1903,188 +2358,1195 @@ subroutine Tlatlon
 
       if (my_task==master_task) then
          write(nu_diag,*) ' '
-         if (nx_block > 5+2*nghost .and. ny_block > 5+2*nghost) then
+!         if (nx_block > 5+2*nghost .and. ny_block > 5+2*nghost) then
          write(nu_diag,*) 'min/max ULON:', y1*rad_to_deg, y2*rad_to_deg
          write(nu_diag,*) 'min/max ULAT:', y3*rad_to_deg, y4*rad_to_deg
-         endif
+!         endif
          write(nu_diag,*) 'min/max TLON:', x1*rad_to_deg, x2*rad_to_deg
          write(nu_diag,*) 'min/max TLAT:', x3*rad_to_deg, x4*rad_to_deg
       endif                     ! my_task
 
+      x1 = global_minval(NLON, distrb_info, nmask)
+      x2 = global_maxval(NLON, distrb_info, nmask)
+      x3 = global_minval(NLAT, distrb_info, nmask)
+      x4 = global_maxval(NLAT, distrb_info, nmask)
+
+      y1 = global_minval(ELON, distrb_info, emask)
+      y2 = global_maxval(ELON, distrb_info, emask)
+      y3 = global_minval(ELAT, distrb_info, emask)
+      y4 = global_maxval(ELAT, distrb_info, emask)
+
+      if (my_task==master_task) then
+         write(nu_diag,*) ' '
+!         if (nx_block > 5+2*nghost .and. ny_block > 5+2*nghost) then
+         write(nu_diag,*) 'min/max NLON:', x1*rad_to_deg, x2*rad_to_deg
+         write(nu_diag,*) 'min/max NLAT:', x3*rad_to_deg, x4*rad_to_deg
+         write(nu_diag,*) 'min/max ELON:', y1*rad_to_deg, y2*rad_to_deg
+         write(nu_diag,*) 'min/max ELAT:', y3*rad_to_deg, y4*rad_to_deg
+!         endif
+      endif                     ! my_task
+
       end subroutine Tlatlon
 
 !=======================================================================
 
-! Transfer vector component from T-cell centers to U-cell centers.
+! Shifts quantities from one grid to another
+! Constructs the shift based on the grid
```

Comment:
I wonder if it makes sense to move all of these grid-shift routines into a new module.